ingest/entryWriter.go

```diff
@@ -191,6 +191,40 @@ func (ew *EntryWriter) ForceAck() error {
 	return ew.forceAckNoLock()
 }
 
+func (ew *EntryWriter) forceAckTimeout(to time.Duration) error {
+	if to <= 0 {
+		//no timeout, just use the regular one
+		return ew.ForceAck()
+	}
+	now := time.Now()
+
+	//setup some timeouts
+	if err := ew.conn.SetWriteTimeout(to); err != nil {
+		return err
+	} else if err := ew.throwAckSync(); err != nil {
+		return err
+	} else if err := ew.conn.ClearWriteTimeout(); err != nil {
+		return err
+	}
+
+	//begin servicing acks with blocking and a read deadline
+	for ew.ecb.Count() > 0 {
+		if time.Since(now) > to {
+			ew.conn.ClearReadTimeout()
+			return ErrTimeout
+		}
+		if err := ew.serviceAcks(true); err != nil {
+			ew.conn.ClearReadTimeout()
+			return err
+		}
+	}
+	if ew.ecb.Count() > 0 {
```

Comment:
Unless there's a race, I don't see how this conditional is ever reached AND true.

Reply:
Agreed but that code is ancient and I didn't want to change more stuff